packages/start-slicemachine/src/lib/setupSentry.ts

```diff
@@ -0,0 +1,70 @@
+import * as Sentry from "@sentry/node";
+
+import semver from "semver";
+import { App, H3Event, sendError } from "h3";
+import { SliceMachineManager } from "@slicemachine/manager";
+import { SENTRY_EXPRESS_DSN } from "../constants";
+
+export const setupSentry = async (
+	manager: SliceMachineManager,
+	app: App,
+): Promise<void> => {
+	!isDevEnv() || process.env.SENTRY_AUTH_TOKEN !== undefined;
+	if (!isSentryEnabled()) {
+		return;
+	}
+
+	const sliceMachineVersion =
+		await manager.versions.getRunningSliceMachineVersion();
+
+	let user;
+	try {
+		user = await manager.user.getProfile();
+	} catch {
+		// not logged in
+		// user stays undefined
+	}
+
+	const config = await manager.project.getSliceMachineConfig();
+
+	const isStableVersion =
+		/^\d+\.\d+\.\d+$/.test(sliceMachineVersion) &&
+		semver.lte("0.1.0", sliceMachineVersion);
+
+	Sentry.init({
+		dsn: process.env.SENTRY_EXPRESS_DSN ?? SENTRY_EXPRESS_DSN,
+		release: sliceMachineVersion,
+		environment: isStableVersion
+			? process.env.NODE_ENV ?? "production"
+			: "alpha",
+	});
+
+	Sentry.setUser({ id: user?.shortId });
```

Comment:
This could be slightly more readable if we wrapped it in an `if` statement instead of using optional chaining. The `if` statement more clearly states that the user is set only if `user` is populated. This is just my personal opinion, so feel free to ignore if you feel otherwise.

```ts
if (user) {
  Sentry.setUser({ id: user.shortId })
}
```